src/main/java/net/fabricmc/fabric/api/event/dimension/AttemptDimensionTeleportCallback.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.dimension;
+
+import net.fabricmc.fabric.api.dimension.EntityPlacer;
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.entity.Entity;
+import net.minecraft.world.dimension.DimensionType;
+
+public interface AttemptDimensionTeleportCallback {
+
+    Event<AttemptDimensionTeleportCallback> EVENT = EventFactory.createArrayBacked(AttemptDimensionTeleportCallback.class, listeners -> (entity, from, to) -> {
+        EventResult currentResult = EventResult.DEFAULT;
+        for(AttemptDimensionTeleportCallback callback : listeners) {
+            EventResult result = callback.placeEntity(entity, from, to);
+            if(result == EventResult.CANCEL_TELEPORT) {
+                return EventResult.CANCEL_TELEPORT;
+            }
+            else if(result != EventResult.DEFAULT) {
+                currentResult = result;
+            }
+        }
+        return currentResult;
+    });
+
+    /**
+     * @return <ul>
+     *     <li>{@link EventResult#SKIP_FURTHER_PROCESSING} if you don't want to have the default {@link EntityPlacer} handle entity placement and portal creation</li>
+     *     <li>{@link EventResult#CANCEL_TELEPORT} if you want to prevent the entity from teleporting at all</li>
+     *     <li>{@link EventResult#DEFAULT} if you want the default {@link EntityPlacer} handle placement of the entity</li>
```

Comment:
Wouldn't this javadoc be more adequate on the enum itself (with a `@see` link) ? Javadoc on the method should rather explain the behaviour of the method itself (when it is called, what it should do...)

Reply:
good point, will fix.

Reply:
fixed.